src/item_tname.cpp

```diff
@@ -520,51 +520,71 @@ std::string category( item const &it, unsigned int /* quantity */,
 // function type that prints an element of tname::segments
 using decl_f_print_segment = std::string( item const &it, unsigned int quantity,
                              segment_bitset const &segments );
-std::unordered_map<tname::segments, decl_f_print_segment *> const segment_map = {
-    { tname::segments::FAULTS, faults },
-    { tname::segments::DIRT, dirt_symbol },
-    { tname::segments::OVERHEAT, overheat_symbol },
-    { tname::segments::FAVORITE_PRE, pre_asterisk },
-    { tname::segments::DURABILITY, durability },
-    { tname::segments::WHEEL_DIAMETER, wheel_diameter },
-    { tname::segments::BURN, burn },
-    { tname::segments::TYPE, label },
-    { tname::segments::CATEGORY, category },
-    { tname::segments::MODS, mods },
-    { tname::segments::CRAFT, craft },
-    { tname::segments::WHITEBLACKLIST, wbl_mark },
-    { tname::segments::CHARGES, noop },
-    { tname::segments::CONTENTS, contents },
-    { tname::segments::FOOD_TRAITS, food_traits },
-    { tname::segments::LOCATION_HINT, location_hint },
-    { tname::segments::ETHEREAL, ethereal },
-    { tname::segments::FOOD_STATUS, food_status },
-    { tname::segments::FOOD_IRRADIATED, food_irradiated },
-    { tname::segments::TEMPERATURE, temperature },
-    { tname::segments::CLOTHING_SIZE, clothing_size },
-    { tname::segments::FILTHY, filthy },
-    { tname::segments::BROKEN, segment_broken },
-    { tname::segments::CBM_STATUS, cbm_status },
-    { tname::segments::UPS, ups },
-    { tname::segments::WETNESS, wetness },
-    { tname::segments::ACTIVE, active },
-    { tname::segments::SEALED, sealed },
-    { tname::segments::FAVORITE_POST, post_asterisk },
-    { tname::segments::WEAPON_MODS, weapon_mods },
-    { tname::segments::RELIC, relic_charges },
-    { tname::segments::LINK, noop },
-    { tname::segments::VARS, vars },
-    { tname::segments::TECHNIQUES, noop },
-    { tname::segments::TAGS, tags },
-};
-
+constexpr size_t num_segments = static_cast<size_t>( tname::segments::last_segment );
+
+constexpr std::array<decl_f_print_segment *, num_segments> get_segs_array()
+{
+    std::array<decl_f_print_segment *, num_segments> arr{};
```

Comment:
Does this correctly default initialize every function pointer with nullptr? I thought std::array requires nested braces eg. `{{}}` to default initialize everything.

Reply:
Because `FOOD_PERISHABLE` doesn't have a printer.

Reply:
Yeah ok gcc and clang properly alert on this.